diff -urN mplayerplug-in.org/Makefile.in mplayerplug-in/Makefile.in
--- mplayerplug-in.org/Makefile.in	2006-05-14 18:52:39.000000000 +0000
+++ mplayerplug-in/Makefile.in	2006-05-14 19:41:24.293212500 +0000
@@ -13,7 +13,7 @@
 CC= @CC@
 CXX= @CXX@
 CFLAGS= -Wall -DXP_UNIX -DMOZ_X11 @MOZPLUG_CFLAGS@ @CFLAGS@ @CPPFLAGS@ @CXXFLAGS@ @CXXFLAGS64@ -Iinclude -fPIC @GECKO_DEFINES@ @GTK_CFLAGS@ @GTHREAD_CFLAGS@ @GTK_DEFINES@
-LDFLAGS= @LDFLAGS@ @LDFLAGS64@ @LIBS@ @XEXT_LIBS@ @XPM_LIBS@ @GTK_LIBS@ @GTHREAD_LIBS@
+LDFLAGS= -pthread @LDFLAGS@ @LDFLAGS64@ @LIBS@ @XEXT_LIBS@ @XPM_LIBS@ @GTK_LIBS@ @GTHREAD_LIBS@
 LIBOBJS= @LIBOBJS@
 OBJ= plugin.o nsScriptablePeer.o npp_gate.o np_entry.o npn_gate.o \
 	plugin-support.o plugin-setup.o plugin-list.o plugin-ui.o \

