diff -urN mplayerplug-in.org/Makefile.in mplayerplug-in/Makefile.in
--- mplayerplug-in.org/Makefile.in	2006-04-27 23:40:06.000000000 +0000
+++ mplayerplug-in/Makefile.in	2006-05-14 14:09:42.285414750 +0000
@@ -27,6 +27,9 @@
 OBJ_RM= plugin.o nsScriptablePeer.o npp_gate.o np_entry.o npn_gate.o \
 	plugin-support.o plugin-setup-rm.o plugin-list.o plugin-ui.o \
 	plugin-threads.o strlcpy.o strlcat.o $(LIBOBJS)
+OBJ_DIVX= plugin.o nsScriptablePeer.o npp_gate.o np_entry.o npn_gate.o \
+	plugin-support.o plugin-setup-divx.o plugin-list.o plugin-ui.o \
+	plugin-threads.o strlcpy.o strlcat.o $(LIBOBJS)
 OBJ_GMP= plugin.o nsScriptablePeer.o npp_gate.o np_entry.o npn_gate.o \
 	plugin-support.o plugin-setup-gmp.o plugin-list.o plugin-ui.o \
 	plugin-threads.o strlcpy.o strlcat.o $(LIBOBJS)
@@ -67,6 +70,10 @@
 	$(CXX) -shared -o $(SHAREDTARGET)-rm.so $(OBJ_RM) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-rm.xpt
 
+$(SHAREDTARGET)-divx: $(OBJ_DIVX)
+	$(CXX) -shared -o $(SHAREDTARGET)-divx.so $(OBJ_DIVX) $(CFLAGS) $(LDFLAGS)
+	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-divx.xpt
+
 $(SHAREDTARGET)-gmp: $(OBJ_GMP)
 	$(CXX) -shared -o $(SHAREDTARGET)-gmp.so $(OBJ_GMP) $(CFLAGS) $(LDFLAGS)
 	cp Source/nsIScriptableMplayerPlugin.xpt mplayerplug-in-gmp.xpt
@@ -95,6 +102,9 @@
 plugin-setup-rm.o: Source/plugin-setup.cpp $(SCRIPTABLEHEADERS)
 	$(CXX) -c -o $@ $(CFLAGS) $(LIBS) -DRM Source/plugin-setup.cpp
 
+plugin-setup-divx.o: Source/plugin-setup.cpp $(SCRIPTABLEHEADERS)
+	$(CXX) -c -o $@ $(CFLAGS) $(LIBS) -DDIVX Source/plugin-setup.cpp
+
 plugin-setup-gmp.o: Source/plugin-setup.cpp $(SCRIPTABLEHEADERS)
 	$(CXX) -c -o $@ $(CFLAGS) $(LIBS) -DGMP Source/plugin-setup.cpp
 
diff -urN mplayerplug-in.org/Source/plugin-setup.cpp mplayerplug-in/Source/plugin-setup.cpp
--- mplayerplug-in.org/Source/plugin-setup.cpp	2006-05-04 15:22:23.000000000 +0000
+++ mplayerplug-in/Source/plugin-setup.cpp	2006-05-14 14:07:58.634937000 +0000
@@ -365,11 +365,6 @@
 	strlcat(MimeTypes,
 		"application/x-nsv-vp3-mp3:nsv:Nullsoft Streaming Video;",
 		sizeof(MimeTypes));
-	// DIVX
-	strlcat(MimeTypes,
-		"video/divx:divx:DivX Media Format;"
-		"video/vnd.divx:divx:DivX Media Format;",
-		sizeof(MimeTypes));
 	// Basic
 	strlcat(MimeTypes,
 		"audio/basic:au,snd:Basic Audio File;"
@@ -389,6 +384,15 @@
 		    sizeof(MimeTypes));
 	}
 #endif
+
+#ifdef DIVX
+        // DIVX
+	strlcat(MimeTypes,
+		"video/divx:divx:DivX Media Format;"
+                "video/vnd.divx:divx:DivX Media Format;",
+                sizeof(MimeTypes));
+#endif
+
 	// Google Video
 #ifdef GMP
 	if (enable_gmp) {
@@ -432,6 +436,10 @@
 	*((const char **) value) = "RealPlayer 9";
 #endif
 
+#ifdef DIVX
+	*((const char **) value) = "DivX\256 Web Player";
+#endif
+
 #ifdef GMP
 	*((const char **) value) = "Google VLC multimedia plugin 1.0";
 #endif
diff -urN mplayerplug-in.org/configure.in mplayerplug-in/configure.in
--- mplayerplug-in.org/configure.in	2006-05-14 06:00:13.000000000 +0000
+++ mplayerplug-in/configure.in	2006-05-14 14:11:18.195408750 +0000
@@ -109,6 +109,9 @@
 AC_ARG_ENABLE(rm,
   AC_HELP_STRING([--enable-rm], [Build with Real Media Support]),
   [if test "$enableval" = ""; then rm=yes; else rm=$enableval; fi],[rm=yes])
+AC_ARG_ENABLE(divx,
+  AC_HELP_STRING([--enable-divx], [Build with DivX Web Player Support]),
+  [if test "$enableval" = ""; then divx=yes; else divx=$enableval; fi],[divx=yes])
 AC_ARG_ENABLE(gmp,
   AC_HELP_STRING([--enable-gmp], [Build with Google Media Support]),
   [if test "$enableval" = ""; then gmp=yes; else gmp=$enableval; fi],[gmp=yes])
@@ -123,6 +126,9 @@
 if test x$rm = xyes; then
    CREATE=" $CREATE mplayerplug-in-rm"
 fi
+if test x$divx = xyes; then
+   CREATE=" $CREATE mplayerplug-in-divx"
+fi
 if test x$gmp = xyes; then
    CREATE=" $CREATE mplayerplug-in-gmp"
 fi
